src/syndrome.jl

```diff
@@ -261,7 +261,7 @@ function berlekamp_massey_decoder!(received::Poly, erasures::AbstractVector, nsy
 
     ## error positions
     errpos = getpositions(Λx)
-    isempty(errpos) && throw(ReedSolomonError())
+    (isempty(errpos) || maximum(errpos) ≥ length(received)) && throw(ReedSolomonError())
```

Comment:
It'll throw `BoundErrors` when `maximum(errpos) ≥ length(received)`, which means the number of errors is too many. It's better to catch the error here.